src/test/java/com/spotify/docker/it/PushIT.java

```diff
@@ -0,0 +1,296 @@
+ */
+public class PushIT {
+
+  private static final int LONG_WAIT_SECONDS = 400;
+  private static final int SECONDS_TO_WAIT_BEFORE_KILL = 120;
+
+  private static final String REGISTRY_IMAGE = "registry:2";
+  private static final String REGISTRY_NAME = "registry";
+
+  private static final String LOCAL_AUTH_EMAIL = "test@example.com";
+  private static final String LOCAL_AUTH_USERNAME = "testuser";
+  private static final String LOCAL_AUTH_PASSWORD = "testpassword";
+  private static final String LOCAL_IMAGE = "localhost:5000/testuser/test-image";
+
+  // Using a dummy individual's test account because organizations
+  // cannot have private repos on Docker Hub.
+  private static final String HUB_AUTH_EMAIL = "dxia+4@spotify.com";
+  private static final String HUB_AUTH_USERNAME = "dxia4";
+  private static final String HUB_AUTH_PASSWORD = "03yDT6Yee4iFaggi";
+  private static final String HUB_PUBLIC_IMAGE =
+      "dxia4/docker-client-test-push-public-image-with-auth";
+  private static final String HUB_PRIVATE_IMAGE =
+      "dxia4/docker-client-test-push-private-image-with-auth";
+
+  private DockerClient client;
+
+  @Rule
+  public final TestName testName = new TestName();
+
+  @BeforeClass
+  public static void before() throws Exception {
+    // Pull the registry image down once before the any test methods in this class run
+    DefaultDockerClient.fromEnv().build().pull(REGISTRY_IMAGE);
+  }
+
+  @Before
+  public void setup() throws Exception {
+    final AuthConfig authConfig = AuthConfig.builder()
+        .email(LOCAL_AUTH_EMAIL)
+        .username(LOCAL_AUTH_USERNAME)
+        .password(LOCAL_AUTH_PASSWORD)
+        .build();
+    client = DefaultDockerClient
+        .fromEnv()
+        .authConfig(authConfig)
+        .build();
+
+    System.out.printf("- %s\n", testName.getMethodName());
+  }
+
+  @Test
+  public void testPushImageToPrivateAuthedRegistryWithoutAuth() throws Exception {
+    final String containerId = startAuthedRegistry(client);
+
+    // Make a DockerClient without AuthConfig
+    final DefaultDockerClient client = DefaultDockerClient.fromEnv().build();
+
+    // Push an image to the private registry and check it fails
+    final String dockerDirectory = Resources.getResource("dockerDirectory").getPath();
+    client.build(Paths.get(dockerDirectory), LOCAL_IMAGE);
+
+    // A bit ugly, but we can't use an @Rule Exception here because we
+    // want to do customized cleanup after the exception is thrown.
+    // We can't just do @Test(expected = ...) because we want to check the exception message.
+    boolean correctExceptionThrown = false;
+    try {
+      client.push(LOCAL_IMAGE);
+    } catch (ImagePushFailedException e) {
+      if (e.getMessage().contains("no basic auth credentials")) {
+        correctExceptionThrown = true;
+      }
+    }
+    assertTrue(correctExceptionThrown);
+
+    stopAndRemoveContainer(client, containerId);
```

Comment:
if you moved the call to `stopAndRemoveContainer` to a `@After` method, then this part here with `correctExceptionThrown` be done away with